Gems/Atom/Feature/Common/Assets/ShaderLib/Atom/Features/Shadow/DirectionalLightShadow.azsli

```diff
@@ -213,20 +220,47 @@ float DirectionalLightShadow::GetVisibilityFromLightPcf()
     static const float PixelMargin = 1.5; // avoiding artifact between cascade levels.
     static const float DepthMargin = 1e-8; // avoiding artifact when near depth bounds.
 
+    bool cascadeFound = false;
+    int currentCascadeIndex = 0;
+
     const uint size = ViewSrg::m_directionalLightShadows[m_lightIndex].m_shadowmapSize;
     const uint cascadeCount = ViewSrg::m_directionalLightShadows[m_lightIndex].m_cascadeCount;
```

Comment:
Didn't check it myself, but is cascadeCount = 1 handled in case a user decides to set it as 1?

Reply:
Good point. I double checked and it should be taken care of with:
                const int nextCascadeIndex = min ( cascadeCount - 1, currentCascadeIndex + 1 ); 
I also tested this just now with a single cascade.